src/dgipy/vcf.py

```diff
@@ -0,0 +1,202 @@
+        if not list(self.interactions["drug"].values):
+            self.applications = "None"
+        else:
+            self.applications = self.search_applications(
+                list(self.interactions["drug"].values)
+            )
+
+        self.gene_info = self.grab_gene_info(self.gene)
+        self.categories = self.grab_categories(self.gene)
+
+    # TODO: These can probably just be simpilifed to direct assignment during __init__ but might be useful this way too
+    def search_interactions(self, gene: str) -> pd.DataFrame:
+        """Search drug-gene interactions for given gene in DGIdb
+
+        :param gene: the name of the gene
+        :return: Dataframe of drug-gene interactions
+        """
+        return dgipy.get_interactions(gene)
+
+    def search_applications(self, drugs: list) -> pd.DataFrame:
+        """Search for drug applications for interaction drugs in DGIdb
+
+        :param drugs: list of drugs from interaction results
+        :return: Dataframe of FDA applications
+        """
+        return dgipy.get_drug_applications(drugs)
+
+    def grab_gene_info(self, gene: str) -> pd.DataFrame:
+        """Grab gene information from DGIdb
+
+        :param gene: gene name
+        :return: Dataframe of gene information
+        """
+        return dgipy.get_gene(gene)
+
+    def grab_categories(self, gene: str) -> pd.DataFrame:
+        """Grab gene categories from DGIdb
+
+        :param gene: gene name
+        :return: Dataframe of gene categories
+        """
+        return dgipy.get_categories(gene)
+
+
+def annotate(filepath: str, contig: str) -> pd.DataFrame:
+    """Map chr,pos pairs from a VCF file to human genes and search DGIdb for drug-gene interactions
+
+    :param filepath: link to a valid VCF file
+    :param contig: specified chromosome (i.e. chr7)
+    :return: Dataframe of drug-gene interactions
+    """
+    # Open VCF file
+    records = _process_vcf(filepath, contig)
+    # Grab records & relevant info (params: chr7)
+    mapped = _ensembl_map(records)  # TODO: modularize mapping
+    # Group records with like-genes
+    grouped = _group_by_name(mapped)
+    # Instance each gene set as a class
+    vcf_results = []
+    for gene in grouped:
+        vcf_results.append(GeneResult(grouped[gene]))
+
+    return vcf_results
+
+
+def _process_vcf(filepath: str, contig: str) -> list:
+    """Grab relevant data for mapping and mutations from starting VCF
+
+    :param filepath: link to valid VCF file
+    :param contig: specified chromosome (i.e. chr7)
+    :return: List of record dicts
+
+    """
+    file = pysam.TabixFile(filepath)
+
+    records = []
+    for record in tqdm(file.fetch(contig)):
+        fields = record.split("\t")
+        entry = {}
+        entry["chromosome"] = fields[0]
+        entry["pos"] = fields[1]
+        entry["ref"] = fields[3]
+        entry["alt"] = fields[4]
+        entry["qual"] = fields[5]
+        entry["filter"] = fields[6]
```

Comment:
small nitpick, but it's a little easier to just assign these all in one fell swoop

```python
entry = {
    "chromosome" = fields[0],
    "pos" = fields[1],
    "ref" = fields[2],
    # and so on
}
```

Reply:
changed